### PR TITLE
feat(registry): Enforce Node network address-related Registry invariant checks

### PR DESCRIPTION
**Pull request by aterga**
This PR enforces Node network address-related Registry invariant checks.

### CODE REVIEW

**Comment by mraszyk** [comment]
> randomly generate some IPs

I mean _deterministicly_ randomly, using some seeded RNG, e.g., similarly to generating node keys.

**Comment by mraszyk** [comment]
Here's a draft PR that should fix the failing test: https://github.com/dfinity/ic/pull/1176

**Review by max-dfinity — APPROVED** [review]

File: rs/registry/canister/src/invariants/endpoint.rs [hunk 1]
```diff
@@ -433,8 +427,10 @@ mod tests {
             .encode_to_vec(),
         );
 
-        // TODO: change to `assert!(check_endpoint_invariants(&snapshot, true).is_err());` after NNS1-2228 is closed.
-        assert!(check_endpoint_invariants(&snapshot, true).is_ok());
+        assert_eq!(check_endpoint_invariants(&snapshot, true), Err(InvariantCheckError {
+            msg: "<TODO: Insert expected error test here>".to_string(),
```
Comment (mraszyk) [inline rs/registry/canister/src/invariants/endpoint.rs:431]: Is this final?

**Review by mraszyk — APPROVED** [review]

File: rs/registry/canister/src/invariants/endpoint.rs (hunk 1 above)
Comment (aterga) [reply rs/registry/canister/src/invariants/endpoint.rs:431]: Nope, not yet

**Review by sasa-tomic — APPROVED** [review]
No concerns from my side. Thanks!

**Pull request merged**